File: src/java/cardealer/com/controller/CarController.java
package cardealer.com.controller;

import cardealer.com.daoImpl.BrandDaoImpl;
import cardealer.com.daoImpl.CarDaoImpl;
import cardealer.com.daoImpl.ModelDaoImpl;
import cardealer.com.model.Brand;
import cardealer.com.model.Car;
import cardealer.com.model.Model;
import java.util.List;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

@Controller
public class CarController {

    CarDaoImpl daoc = new CarDaoImpl();
    ModelDaoImpl daom = new ModelDaoImpl();
    BrandDaoImpl daob = new BrandDaoImpl();

     ////Get list of brands to post.   
    @RequestMapping(value = "/car", method = RequestMethod.GET)
    public @ResponseBody List<Brand> getBrands() {
        List<Brand> brands = daob.showBrands();
        return brands;

    }
    
    //Get list of models to post.
    @RequestMapping(value = "/models", method = RequestMethod.GET)
    public @ResponseBody List<Model> getModels() {
        List<Model> models = daom.showModels();
        return models;
    }
    
    //Create a new car.
    @RequestMapping(value = "/car", method = RequestMethod.POST)
    public @ResponseBody Car createCar(@RequestBody Car car){
        daoc.createCar(car);
        return car;
    }
    
    //Show cars registered.
    @RequestMapping(value = "/listcars", method = RequestMethod.GET)
    public @ResponseBody List<Car> getCars(){
        List<Car> cars = daoc.showCars();
        return cars;
    }
    
    //Update a car.
    @RequestMapping(value = "/updateCar", method = RequestMethod.PUT)
    public @ResponseBody Car updateCar(@RequestBody Car car) {
        daoc.updateCar(car);
        return car;
    }

    //Delete a car.
    @RequestMapping(value = "/deleteCar", method = RequestMethod.DELETE)
    public @ResponseBody Car deleteCar(@RequestBody Car car) {
        daoc.deleteCar(car);
        return car;
    }

}

File: web/webapp/resources/static/app/carController.js
app.controller('carCtrl', function ($scope, $http, $location, $localStorage, ngDialog) {
    //Creating variables.
    $scope.car = {brand: {},
        model: {}
    };

    $scope.brands = [];
    $scope.models = [{
            brand: {}
        }];

    $scope.modelsToPost = [{
            brand: {}
        }];

    $scope.listCars = [];
    $scope.showModels = true;


    //Make selection of models depending on which brand was chosen.
    $scope.change = function () {
        $scope.modelsToPost = [];
        angular.forEach($scope.models, function (item) {
            console.log(item);
            if ($scope.car.brand.idBrand == item.brand.idBrand) {
                $scope.modelsToPost.push(item);
            }
        });
        $scope.showModels = false;
    };

    //Get brands to post.
    $http({
        method: 'GET',
        url: 'car',
        data: $scope.brands,
        headers: {'Content-Type': 'application/json'}
    }).success(function (data) {
        $scope.brands = data;
    });

    //Get models to post
    $http({
        method: 'GET',
        url: 'models',
        data: $scope.models,
        headers: {'Content-Type': 'application/json'}
    }).success(function (data) {
        $scope.models = data;
    });

    //Create a new car.
    $scope.createCar = function () {

        $http({
            method: 'POST',
            url: 'car',
            data: $scope.car,
            headers: {'Content-Type': 'application/json'}
        }).success(function (data) {
            $location.path('/listcars');
        });
    };

    //Show cars created.
    $scope.show = "true";
    $scope.hide = "true";

    $http({
        method: 'GET',
        url: 'listcars',
        data: $scope.listCars,
        headers: {'Content-Type': 'application/json'}
    }).success(function (data) {
        $scope.listCars = data;
    });

    //Ediit car
    $scope.editCar = function (idCar) {
        $scope.show = "false";
        $scope.hide = "false";
        $scope.hideObj = "true";
        $scope.showObj = "true";

        $scope.idCar = idCar;
    };

    //Back to normal.
    $scope.undoEdit = function () {
        $scope.show = "true";
        $scope.hide = "true";
        $scope.hideObj = "false";
        $scope.showObj = "false";

        $scope.idCar = 0;
    };

    //Save changes made in the selected field.
    $scope.saveCar = function (car) {
        $http({
            method: 'PUT',
            url: 'updateCar',
            data: car,
            headers: {'Content-Type': 'application/json'}
        }).success(function (data) {
            ngDialog.open({
                scope: $scope,
                template: 'webapp/resources/static/img/modalSuccess.html',
                className: 'ngdialog-theme-default',
                width: 650,
                lain: true
            });
            $scope.show = "true";
            $scope.hide = "true";
            $scope.hideObj = "false";
            $scope.showObj = "false";

            $scope.idCar = 0;
            $location.path('/listcars');
        });
    };

    //Delete car.
    $scope.deleteCar = function (car) {
        ngDialog.openConfirm({
            template: 'webapp/resources/static/img/sure.html',
            className: 'ngdialog-theme-default'
        }).then(function () {
            $http({
                method: 'DELETE',
                url: 'deleteCar',
                data: car,
                headers: {'Content-Type': 'application/json'}
            }).success(function (car) {
                //Update list. Again again it's not really what I wanted to do.
                $http({
                    method: 'GET',
                    url: 'listcars',
                    data: $scope.listCars,
                    headers: {'Content-Type': 'application/json'}
                }).success(function (data) {
                    $scope.listCars = data;
                });

            });
        }, function (reason) {

        });
    };
});